lib/superset/dashboard/datasets/list.rb

```diff
@@ -36,13 +37,34 @@ def schemas
         end
 
         def datasets_details
-          result.map do |details|
+          chart_datasets = result.map do |details|
             details.slice('id', 'datasource_name', 'schema', 'sql').merge('database' => details['database'].slice('id', 'name', 'backend')).with_indifferent_access
           end
+          return chart_datasets unless include_filter_datasets
+          chart_dataset_ids = chart_datasets.map{|d| d['id'] }
+          filter_dataset_ids_not_used_in_charts = filter_dataset_ids - chart_dataset_ids
+          return chart_datasets if filter_dataset_ids_not_used_in_charts.empty?
+          dashboard_datasets = chart_datasets + filter_datasets(filter_dataset_ids_not_used_in_charts)
         end
 
         private
 
+        def filter_dataset_ids
+          @filter_dataset_ids ||= Dashboard::Filters::List.new(id).perform
```

Comment:
See comment below .. we could memoize a dashboard object
and use it for the `dashboard.title` method as well as `dashboard.filter_configuration`

Just trying to minimise the number of API calls we use

      def dashboard
        @dashboard ||= Superset::Dashboard::Get.new(dashboard_id)
      end